.pre-commit-config.yaml

```diff
@@ -1,43 +1,27 @@
 repos:
-  - repo: https://github.com/adamchainz/blacken-docs
-    rev: "v1.12.1"
-    hooks:
-      - id: blacken-docs
-        additional_dependencies:
-          - black==22.12.0
   - repo: https://github.com/charliermarsh/ruff-pre-commit
     # Ruff version.
-    rev: "v0.0.265"
+    rev: "v0.3.0"
     hooks:
       - id: ruff
         # We use `--no-cache` to ensure that we always run based on what the
         # pre-commit tool provides us.
         args: [ --fix, --exit-non-zero-on-fix, --no-cache ]
+      - id: ruff-format
+        # We use `--no-cache` to ensure that we always run based on what the
+        # pre-commit tool provides us.
+        args: [ --no-cache ]
   - repo: https://github.com/pycqa/pylint
     rev: v2.14.5
     hooks:
       - id: pylint
-  - repo: https://github.com/hadialqattan/pycln
-    rev: v2.4.0
-    hooks:
-      - id: pycln
-        args: [ --config=pyproject.toml ]
-  - repo: https://github.com/PyCQA/autoflake
-    rev: v2.1.1
-    hooks:
-      - id: autoflake
-  - repo: https://github.com/pre-commit/pygrep-hooks
-    rev: v1.10.0
-    hooks:
-      - id: python-use-type-annotations
   - repo: https://github.com/pycqa/flake8
     rev: 6.0.0
     hooks:
       - id: flake8
-        additional_dependencies: [ flake8-annotations ]
         args:
           - --max-line-length=120
-          - --ignore=ANN101,ANN102 # Ignoring annotation for self and cls
+          - --ignore=ANN101,ANN102,ANN204 # Ignoring annotation for self, cls, and special methods
```

Comment:
Let's remove both of these unless there are specific checks you are partial to, they should be mostly covered by our `ruff` checks.